### DISCUSSION TITLE
v0.90.7-beta

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Trilium#6046] [Releases]
⚠️ Important notes
------------------

*   0.90.5-beta & 0.90.6-beta have an incompatibility with older sync servers if using mind maps, causing them to get turned to a file. If you are impacted by this, see [mindmap note suddenly show with file note](https://github.com/TriliumNext/Notes/issues/467#issuecomment-2402853189).
*   Sync version has been increased from 32 to 33 to prevent issues with mindmap notes. This means that the server also needs to be updated to this version, otherwise sync will be refused.

🐞 Bugfixes
-----------

*   [Insert Math appears to be broken](https://github.com/TriliumNext/Notes/issues/297)
*   [Override the z-index of Bootstrap tooltips](https://github.com/TriliumNext/Notes/pull/472) by [dousha](https://github.com/dousha)
*   [Clicking an entry in the calendar opens the note twice](https://github.com/TriliumNext/Notes/issues/417)
*   [mindmap note suddenly show with file note](https://github.com/TriliumNext/Notes/issues/467)
*   [Import md with jpeg: broken jpeg](https://github.com/TriliumNext/Notes/pull/458) by [`perfectra1n`](https://github.com/perfectra1n)

✨ Improvements
--------------

*   [Icon improvements](https://github.com/TriliumNext/Notes/pull/392) by `@SiriusXT`.
*   [Open in file manager of Trilium's data directory](https://github.com/TriliumNext/Notes/pull/389) by `@SiriusXT`.
*   [i18n Fallback to English for null characters](https://github.com/TriliumNext/Notes/pull/414) by `@SiriusXT`.
*   [Use matrix strategy for CI/CD](https://github.com/TriliumNext/Notes/pull/402#top) (Alpine is back!) by `perfectra1n`.
*   [Floating buttons can be displayed again after being closed](https://github.com/TriliumNext/Notes/pull/432) by `@SiriusXT`.
*   [Revisions number limit](https://github.com/TriliumNext/Notes/pull/391) by [SiriusXT](https://github.com/SiriusXT).
*   [Searching with orderBy doesn't work with dated attributes](https://github.com/TriliumNext/Notes/pull/451) by [jaimeferj](https://github.com/jaimeferj).
*   [Use the electron Clipboard module when using "Copy image to clipboard"](https://github.com/TriliumNext/Notes/pull/452) by [`perfectra1n`](https://github.com/perfectra1n).
*   [fix blurry PWA icon](https://github.com/TriliumNext/Notes/pull/465) by [quantum5](https://github.com/quantum5)

🌍 Internationalization
-----------------------

*   Translated more of the client (e.g. note types, tree context menu, switches in the ribbon).
*   Started translating the server (e.g. first setup, keyboard actions, shared pages).
*   Improved coverage for Spanish by [hasecilu](https://github.com/hasecilu).
*   Improved coverage for Chinese by [Nriver](https://github.com/Nriver/).
*   Improved coverage for Romanian.

Library updates
---------------

*   [Upgrade bootstrap from v4 to v5](https://github.com/TriliumNext/Notes/pull/381) by @JYC333
    *   This is a heavy UI library update, so there might be quirks here and there.
    *   Various improvements and bugfixes by @JYC333 and @SiriusXT
*   Mermaid: 10.9.1 → 11.3.0
*   Mind Elixir: 4.0.5 → 4.1.5
*   JSDOM: 24.1.1 → 25.0.0
*   Day.js: 1.11.12 → 1.11.13
*   ESLint: 9.9.0 → 9.10.0
*   Marked: 13.0.2 → 14.1.2
*   Ini: 4.1.3 -> 5.0.0
*   Axios: 1.7.2 → 1.7.7
*   i18next: 23.14.0 → 23.15.2
*   express 4.19.2 → 4.21.0
*   better-sqlite3: 11.1.2 → 11.3.0
*   and other development tools.

<hr /><em>This discussion was created from the release <a href='https://github.com/TriliumNext/Notes/releases/tag/v0.90.7-beta'>v0.90.7-beta</a>.</em>

**Comment by maphew** [comment]
Is changing the db sync version the only way to handle the new Mind Map notes type? even if it's not used?
A new DB sync version means the pool of testers is now limited only to those using standalone desktop or are able/willing to update their servers to run a beta.

**Reply by eliandoran** [reply]
> Is changing the db sync version the only way to handle the new Mind Map notes type? even if it's not used?

In the previous beta version it would be possible to test the new mind map note type without the need to update the server as well. The problem is that the consistency check of an older server would not recognize the new note type and turn it into a file, which can be quite problematic to fix (see https://github.com/TriliumNext/Notes/issues/467#issuecomment-2402853189).

Bumping the sync server does fix this issue. Theoretically we should have bumped the DB version as well, since downgrading to an older version that does not have the mind map will also trigger the issue without the need of a sync server, but I thought it was too much.

The only other way is to modify the consistency checker to handle unknown note types in a different way. We'd have to think it through, especially if we want to introduce a plugin system.

**Reply by maphew** [reply]
Thanks for the added context.

> The only other way is to modify the consistency checker to handle unknown note types in a different way. We'd have to think it through, especially if we want to introduce a plugin system.

So it sounds to me like this will be necessary regardless, though perhaps delayed until introducing a plugin system becomes the active development priority.

**Comment by capi** [comment]
If the sync (and soon-is maybe the DB?) becomes incompatible with the original Trilium, maybe do more than "just" incrementing the sync version, in case zadam does do a newer release at a later point in time?

It could become quite nasty if zadam does an update and people later want to switch over from Trilium to Trilium Next. I myself I am actively following this project and I really appreciate all you are doing, but currently I am still running zadam's version on the server and waiting for this to come out of Beta before doing the switch-over.

**Reply by perfectra1n** [reply]
There's a [v0.90.4 release](https://github.com/TriliumNext/Notes/releases/tag/v0.90.4) that doesn't increment the DB version if you wish to use it :)

But v0.90.8 is out now!

**Reply by capi** [reply]
Yes, I saw that and that's awesome! My point was more that we are now breaking away and actually should start to not sync against the old zaadam backends any more (recognize if the sync server is Trilium or TriliumNext) in order to avoid issues _in case_ Trilium updates DB and/or sync protocol versions as well.

**Reply by eliandoran** [reply]
@capi , in those circumstances we'll most likely bump the sync version even further like we did with the application version. We switched directly to v0.90+ in order to avoid overlapping with zadam's version.

Honestly I don't expect the sync version to increase by much (if at all) during the maintenance phase. Generally the sync version is increased when we add new features only.
